### PR TITLE
 code cleaned up of runtime module

### PR DESCRIPTION
**Pull request by bnazare**
Similar to #108 , a bit of code clean-up in module "runtime". Namely:
- Removed unused code
- Added final modifiers where applicable
- Sprinkled in some lambda functions in place of anonymous classes
- Fixed strings comparisons
- Added a metric ton of `@Override` annotations
- Replaced `Exception` in catch clauses with more specific exceptions
- Fixed some JavaDocs

Sorry if this touches too many files, at first I meant to split it into several PRs. However, I might have to leave this stuff for a while so I'm dumping all I've got for now :).

### CODE REVIEW

File: runtime/src/main/java/io/quarkiverse/cxf/CXFClientInfos.java [hunk 1]
```diff
@@ -28,7 +30,7 @@ public static CXFClientInfo fromSei(String sei) {
 
     public CXFClientInfo getClientInfoBySei(String sei) {
         for (CXFClientInfo clientInfo : cxfClientInfos) {
-            if (clientInfo.getSei() == sei) {
+            if (Objects.equals(clientInfo.getSei(), sei)) {
```
Comment (bnazare) [inline runtime/src/main/java/io/quarkiverse/cxf/CXFClientInfos.java:33]: The original code yielded true when both strings were null, but I'm not sure if that was actually a requirement. This code maintains the same behaviour but let me know if I should change it to a simple `String.equals()`.

File: runtime/src/main/java/io/quarkiverse/cxf/CXFQuarkusServlet.java [hunk 2]
```diff
@@ -57,7 +57,7 @@ private Object getInstance(String className) {
         Class<?> classObj = loadClass(className);
         try {
             return classObj.getConstructor().newInstance();
-        } catch (Exception e) {
+        } catch (ReflectiveOperationException | RuntimeException e) {
```
Comment (bnazare) [inline runtime/src/main/java/io/quarkiverse/cxf/CXFQuarkusServlet.java:60]: The original code was catching `Exception` which means it was implicitly catching `RuntimeException`s too. For that reason I kept `RuntimeException` in the revised catch clause. This happens in a few other places too.
Ideally, catch clauses with `RuntimeException`s should be removed except where strictly necessary, but I didn't want to make that call at this point.

File: runtime/src/main/java/io/quarkiverse/cxf/transport/VertxServletOutputStream.java [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
 public class VertxServletOutputStream extends ServletOutputStream {
 
-    private HttpServerRequest request;
+    private final HttpServerRequest request;
     protected HttpServerResponse response;
     private ByteBuf pooledBuffer;
     private long written;
```
Comment (bnazare) [inline runtime/src/main/java/io/quarkiverse/cxf/transport/VertxServletOutputStream.java:26]: Fields `written` and `finished` are written to but never read which means they can be removed. Let me know if you want me to do this.

Comment (bnazare) [inline runtime/src/main/java/io/quarkiverse/cxf/transport/VertxServletOutputStream.java:26]: Fields `written` and `finished` are written to but never read which means they can be removed. Let me know if you want me to do this.

File: runtime/src/main/java/io/quarkiverse/cxf/CXFQuarkusServlet.java [hunk 4]
```diff
@@ -35,6 +34,7 @@
 public class CXFQuarkusServlet extends CXFNonSpringServlet {
 
     private static final Logger LOGGER = Logger.getLogger(CXFQuarkusServlet.class);
+    private static final long serialVersionUID = 1371666057918777501L;
```
Comment (dufoli) [inline runtime/src/main/java/io/quarkiverse/cxf/CXFQuarkusServlet.java:37]: I think it was added by mistake...

File: runtime/src/main/java/io/quarkiverse/cxf/CXFQuarkusServlet.java [hunk 5]
```diff
@@ -35,6 +34,7 @@
 public class CXFQuarkusServlet extends CXFNonSpringServlet {
```
Comment (dufoli) [inline runtime/src/main/java/io/quarkiverse/cxf/CXFQuarkusServlet.java:34]: indeed this class can be totaly removed because I use vertx route now ;-)

**Review by dufoli — COMMENTED** [review]
sorry I merge it but servlet can be totaly removed indeed and you add a serialVersionUID field

**Comment by dufoli** [comment]
@bnazare I forget to thank you for this patch ;-)

**Pull request merged**